Backpack/OverlayView/Classes/BPKOverlayView.m

```diff
@@ -0,0 +1,192 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2018-2020 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#import "BPKOverlayView.h"
+
+#import <Backpack/Color.h>
+#import <Backpack/Common.h>
+#import <Backpack/Duration.h>
+#import <Backpack/Radii.h>
+#import <Backpack/Shadow.h>
+#import <Backpack/Spacing.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+@interface BPKOverlayView ()
+
+@property(nonatomic, strong, nullable) CALayer *tintLayer;
```

Comment:
This feels a bit confusing. `BPKOverlayView` now owns a layer with `strong` memory semantics that is in fact a sublayer of a subview of `BPKOverlayView`, would it make more sense to move this ownership and responsbility into a new class and make `backgroundView` an instance of it?

Reply:
Refactor done